website/source/guides/security/vault-pki-integration.html.md

```diff
@@ -0,0 +1,550 @@
+read more about Vault policies [here][policies].
+
+Write the policy we just created into Vault:
+
+```
+$ vault policy write tls-policy tls-policy.hcl
+Success! Uploaded policy: tls-policy
+```
+
+### Step 8: Generate a Token based on `tls-policy`
+
+Create a token based on `tls-policy` with the following command:
+
+```
+$ vault token create -policy="tls-policy" -ttl=24h
+```
+
+If the command is successful, you will see output similar to the following:
+
+```shell
+Key                  Value
+---                  -----
+token                s.xafiYzh7MCMotHLu2d35hepR
+token_accessor       9vj7q5nnF53JAcTyxvccpAZK
+token_duration       24h
+token_renewable      true
+token_policies       ["default" "tls-policy"]
+identity_policies    []
+policies             ["default" "tls-policy"]
+```
+
+Make a note of this token as you will need it in the upcoming steps.
+
+### Step 9: Configure Consul Template on All Nodes
+
+If you are using the AWS environment provided in this guide, you already have
+[Consul Template][consul-template-github] installed on all nodes. If you are
+using your own environment, please make sure Consul Template is installed. You
+can download it [here][ct-download].
+
+Provide the token you created in [Step
+8](#step-8-generate-a-token-based-on-tls-policy) to the Consul Template
+configuration file located at `/etc/consul-template.d/consul-template.hcl`. You
+will also need to specify the [template stanza][ct-template-stanza] so you can
+render each of the following on your nodes at the specified location from the
+provided templates (the actual templates will be provided in the next step):
+
+* Node certificate
+* Node private key
+* CA public certificate
+
+We will also specify the template stanza to create certs and keys for the Nomad
+CLI (which defaults to HTTP but will need to use HTTPS once once TLS is enabled
+in our cluster).
+
+Your `consul-template.hcl` configuration file should look similar to the
+following (you will need to provide this to each node in the cluster):
+
+```
+vault {
+  address      = "http://active.vault.service.consul:8200"
+  token        = "s.xafiYzh7MCMotHLu2d35hepR"
+  grace        = "1s"
+  unwrap_token = false
+  renew_token  = true
+}
+
+syslog {
+  enabled  = true
+  facility = "LOCAL5"
+}
+
+template {
+  source      = "/opt/nomad/templates/agent.crt.tpl"
+  destination = "/opt/nomad/certs/agent.crt"
+  command     = "pkill -SIGHUP nomad"
+}
+
+template {
+  source      = "/opt/nomad/templates/agent.key.tpl"
+  destination = "/opt/nomad/certs/agent.key"
+  command     = "pkill -SIGHUP nomad"
+}
+
+template {
+  source      = "/opt/nomad/templates/ca.crt.tpl"
+  destination = "/opt/nomad/certs/ca.crt"
+  command     = "pkill -SIGHUP nomad"
+}
+
+template {
+  source      = "/opt/nomad/templates/cli.crt.tpl"
+  destination = "/opt/nomad/certs/cli.crt"
+}
+
+template {
+  source      = "/opt/nomad/templates/cli.key.tpl"
+  destination = "/opt/nomad/certs/cli.key"
```

Comment:
Giving agent and CLI certs the same access permissions defeats the purpose of different certificates for each. The CLI cert is to prevent operators from having access to the agent certificate.

Perhaps put the agent and CLI certificates in two different directories with the agent certificate directory having the permissions: 0700 (owner only access - root in this case).

That allows any system user to use the CLI but only users with access to root can use the agent certificate.

Reply:
Thanks for catching that. Consul Template actually uses the *perms* options to control this, so I'll make sure to incorporate that.